test/Streamly/Test/Prelude/Serial.hs

```diff
@@ -106,55 +107,151 @@ splitOnSuffixSeq = do
         S.toList
              $ IS.splitOnSuffixSeq (A.fromList pat) (FL.toList) (S.fromList xs)
 
+seqSplitterProperties ::
+       forall a. (Arbitrary a, Eq a, Show a, Storable a, Enum a)
+    => a
+    -> String
+    -> Spec
+seqSplitterProperties sep desc = do
+    describe (desc <> " splitOnSeq/intercalate")
+        $ do
+            -- Empty seperator
+            intercalateSplitEqId 0
+            concatSplitIntercalateEqConcat splitOnSeq_ intercalate 0
```

Comment:
In the same tests we could add `Streamly.intercalate` and `Streamly.intercalateSuffix` as well. Do not remove the list intercalate, keep that too, streamly intercalate should also provide the same results as list intercalate.

Reply:
Why would we want to test `Streamly.intercalate` here?

Reply:
Ideally we should be using Streamly.intercalate instead of List.intercalate here because we want to test that Streamly.splitOnSeq and Streamly.intercalate are inverses. But its ok to use List.intercalate to test against an existing implementation of intercalate.